holoviews/plotting/bokeh/stats.py

```diff
@@ -35,3 +47,377 @@ class BivariatePlot(PolygonPlot):
+                    else:
+                        r2_data[kd_name] = [k]
+                if vdim in r1_data:
+                    r1_data[vdim].append(q2)
+                else:
+                    r1_data[vdim] = [q2]
+                if vdim in r2_data:
+                    r2_data[vdim].append(q2)
+                else:
+                    r2_data[vdim] = [q2]
+
+        # Define combined data and mappings
+        bar_glyph = 'hbar' if self.invert_axes else 'vbar'
+        data = {
+            bar_glyph+'_1': r1_data, bar_glyph+'_2': r2_data, 'segment_1': s1_data,
+            'segment_2': s2_data, 'rect_1': w1_data, 'rect_2': w2_data,
+            'circle_1': out_data
+        }
+        mapping = {
+            bar_glyph+'_1': vbar_map, bar_glyph+'_2': vbar2_map, 'segment_1': seg_map,
+            'segment_2': seg_map, 'rect_1': whisk_map, 'rect_2': whisk_map,
+            'circle_1': out_map
+        }
+
+        # Cast data to arrays to take advantage of base64 encoding
+        for gdata in [r1_data, r2_data, s1_data, s2_data, w1_data, w2_data, out_data]:
+            for k, values in gdata.items():
+                gdata[k] = np.array(values)
+
+        # Return if not grouped
+        if not element.kdims:
+            return data, mapping, style
+
+        # Define color dimension and data
+        if cidx is None or cidx>=element.ndims:
+            cdim = Dimension('index')
+        else:
+            r1_data[dimension_sanitizer(cdim.name)] = factors
+            r2_data[dimension_sanitizer(cdim.name)] = factors
+            factors = list(unique_iterator(factors))
+
+        # Get colors and define categorical colormapper
+        cname = dimension_sanitizer(cdim.name)
+        cmap = style.get('cmap')
+        if cmap is None:
+            cycle_style = self.lookup_options(element, 'style')
+            styles = cycle_style.max_cycles(len(factors))
+            colors = [styles[i].get('box_color', styles[i]['box_fill_color'])
+                      for i in range(len(factors))]
+            colors = [rgb2hex(c) if isinstance(c, tuple) else c for c in colors]
+        else:
+            colors = None
+        mapper = self._get_colormapper(cdim, element, ranges, style, factors, colors)
+        vbar_map['fill_color'] = {'field': cname, 'transform': mapper}
+        vbar2_map['fill_color'] = {'field': cname, 'transform': mapper}
+        if self.show_legend:
+            vbar_map['legend'] = cdim.name
+
+        return data, mapping, style
+
+
+
+class ViolinPlot(BoxWhiskerPlot):
+
+    bandwidth = param.Number(default=None, doc="""
+        Allows supplying explicit bandwidth value rather than relying
+        on scott or silverman method.""")
+
+    clip = param.Boolean(default=True, doc="""
+        Whether to clip the violins at the max and min of the data.""")
+
+    cut = param.Number(default=5, doc="""
+        Draw the estimate to cut * bw from the extreme data points.""")
+
```

Comment:
Looks like the matplotlib implementation doesn't have the ``clip`` and ``cut`` plot options?

Reply:
Right, the matplotlib violin implementation doesn't expose it for some reason, it just always clips.

Reply:
Ok. That is fine as long as these defaults match the behavior of the matplotlib version.

Reply:
It does.